src/csharp/Microsoft.Spark/Sql/RowConstructor.cs

```diff
@@ -16,74 +18,89 @@ namespace Microsoft.Spark.Sql
     internal sealed class RowConstructor : IObjectConstructor
     {
         /// <summary>
-        /// Schema of the rows being received. Note that this is thread local variable
+        /// Cache the schema of the rows being received. Note that this is thread local variable
         /// because one RowConstructor object is registered to the Unpickler and there
         /// could be multiple threads unpickling the data using the same object registered.
         /// </summary>
         [ThreadStatic]
-        private static StructType s_currentSchema;
+        private static ConcurrentDictionary<string, StructType> s_schemaCache;
```

Comment:
Any reason this is` ConcurrentDictionary` AND `ThreadStatic`?

Reply:
changing to `Dictionary`